src/main/groovy/org/jetbrains/intellij/tasks/RunPluginVerifierTask.groovy

```diff
@@ -0,0 +1,232 @@
+        this.pluginsToCheckAllBuilds = pluginsToCheckAllBuilds
+    }
+
+    @Input
+    String getPluginsToCheckLastBuilds() {
+        return pluginsToCheckLastBuilds
+    }
+
+    void setPluginsToCheckLastBuilds(String pluginsToCheckLastBuilds) {
+        this.pluginsToCheckLastBuilds = pluginsToCheckLastBuilds
+    }
+
+    @Input
+    String getTeamCity() {
+        return teamCity
+    }
+
+    void setTeamCity(String teamCity) {
+        this.teamCity = teamCity
+    }
+
+    @Input
+    String getTcGrouping() {
+        return tcGrouping
+    }
+
+    void setTcGrouping(String tcGrouping) {
+        this.tcGrouping = tcGrouping
+    }
+
+    @Input
+    String getExcludedPluginsFile() {
+        return excludedPluginsFile
+    }
+
+    void setExcludedPluginsFile(String excludedPluginsFile) {
+        this.excludedPluginsFile = excludedPluginsFile
+    }
+
+    @Input
+    String getDumpBrokenPluginList() {
+        return dumpBrokenPluginList
+    }
+
+    void setDumpBrokenPluginList(String dumpBrokenPluginList) {
+        this.dumpBrokenPluginList = dumpBrokenPluginList
+    }
+
+    @Input
+    String getPluginsToCheckFile() {
+        return pluginsToCheckFile
+    }
+
+    void setPluginsToCheckFile(String pluginsToCheckFile) {
+        this.pluginsToCheckFile = pluginsToCheckFile
+    }
+
+    @Input
+    String getSubsystemsToCheck() {
+        return subsystemsToCheck
+    }
+
+    void setSubsystemsToCheck(String subsystemsToCheck) {
+        this.subsystemsToCheck = subsystemsToCheck
+    }
+
+    @TaskAction
+    void runPluginVerifier() {
+        def extension = project.extensions.findByType(IntelliJPluginExtension)
+        def pluginFileName = "${extension.pluginName}-${project.version}"
+
+        if (!project.file("${project.buildDir}/distributions/${pluginFileName}.zip").exists()) {
+            throw new IllegalStateException("Plugin file $pluginFileName does not exist.")
+        }
+
+        if (ides.isEmpty()) {
+            ides.add("${extension.type}-${extension.version}")
+        }
+
+        def resolver = new IdeaDependencyManager(extension.intellijRepo ?: IntelliJPlugin.DEFAULT_INTELLIJ_REPO)
+        def verifierPath = getVerifierPath()
+
+        def verifierArgs = ["check-plugin"]
+        verifierArgs += getOptions()
+        verifierArgs += ["${project.buildDir}/distributions/${pluginFileName}.zip"]
+        verifierArgs += ides.collect {
+            def (String type, String version) = it.split("-")
+            def dependency = resolver.resolveRemote(project, version, type, false)
+            return dependency.classes.absolutePath
+        }
+
+        project.javaexec {
+            classpath = project.files(verifierPath)
+            main = "com.jetbrains.pluginverifier.PluginVerifierMain"
+            args = verifierArgs
+        }
+    }
+
+    private String getVerifierPath() {
+        project.repositories.maven { it.url = IntelliJPlugin.DEFAULT_INTELLIJ_PLUGIN_SERVICE }
```

Comment:
probably it makes sense to remove repo after downloading

Reply:
Good point, done.